### PR TITLE
Protect foreign key constraint between `imagelist_images` and `imagelist`

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

Since migration 0002 changes the primary key on the `ImageList` model, it breaks the foreign key constraint that refers to that column. This PR merges 0002 into 0001 so that the constraint is set **after** the changes to the primary key.

Credits to @AetherUnbound for pointing this bug out in prod.

## Testing Instructions

1. Migrate the database
    ```sh
    docker-compose exec -T web python manage.py migrate
    ```
1. Describe the `imagelist_images` table.
    ```sh
    docker-compose exec -T db psql -U deploy -d openledger <<-EOF
    \d+ imagelist_images;
    EOF
    ```
1. Ensure 2 foreign key constraints:
    - `FOREIGN KEY (image_id) REFERENCES image(id)`
    - `FOREIGN KEY (imagelist_id) REFERENCES imagelist(id)`

## ⚠️ Production notice

Since 0002 is already a performed migration, removing the corresponding file might interfere with rollbacks etc. To prevent potential future problems, remove the entry from the `django_migrations` table.

```sql
DELETE FROM 
  django_migrations
WHERE
  app = 'api' 
  AND name = '0002_auto_20180723_1737';
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ingestion_server/test/unit_test.py [hunk 1]
```diff
@@ -140,9 +140,9 @@ def test_url_protocol_fix():
         result = CleanupFunctions.cleanup_url(bad_url, tls_support_cache)
         expected = "'https://flickr.com'"
 
-        bad_http = "neverssl.com"
+        bad_http = "baidu.com"
         result_http = CleanupFunctions.cleanup_url(bad_http, tls_support_cache)
-        expected_http = "'http://neverssl.com'"
+        expected_http = "'http://baidu.com'"
```
Comment (sarayourfriend) [inline ingestion_server/test/unit_test.py:145]: Was this change intentional?

Comment (zackkrida) [inline ingestion_server/test/unit_test.py:145]: I can speculate as to why, but what's the reason for the domain switch here?

**Review by sarayourfriend — COMMENTED** [review]
Is there any way to fix this that doesn't involve changing the migration history? If not this LGTM but I think leaving the migration history as immutable is preferrable to modifying it.

File: openverse_api/catalog/api/migrations/0003_image_view_count.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 class Migration(migrations.Migration):
 
     dependencies = [
-        ('api', '0002_auto_20180723_1737'),
+        ('api', '0001_initial'),  # 0002 was merged into 0001 to protect FK constraint
```
Comment (zackkrida) [inline openverse_api/catalog/api/migrations/0003_image_view_count.py:9]: 👍

**Review by zackkrida — APPROVED** [review]
Thank you @dhruvkb and @AetherUnbound, this is great.

File: ingestion_server/test/unit_test.py (hunk 1 above)
Comment (dhruvkb) [reply ingestion_server/test/unit_test.py:145]: NeverSSL now supports SSL ☹️

Comment (dhruvkb) [reply ingestion_server/test/unit_test.py:145]: NeverSSL now supports SSL ☹️ I read that Baidu doesn't. The tests are passing locally, not sure why they're failing on CI.

Comment (zackkrida) [reply ingestion_server/test/unit_test.py:145]: Another one I know is http://mitm.cool (`man in the middle` lol)

Comment (sarayourfriend) [reply ingestion_server/test/unit_test.py:145]: Do these make an actual HTTP request during the tests?

If so, depending on the behavior of a 3rd party server for the validity of our test suite is not good. If we need to test a non-SSL http request, then we should mock the request using something like VCR or nock (whatever the Python equivalent of those is).

Comment (dhruvkb) [reply ingestion_server/test/unit_test.py]: Updated the test to use mocks in 2181e32.

Comment (dhruvkb) [reply ingestion_server/test/unit_test.py]: That's cool but mocking is the stablest solution.

File: ingestion_server/test/unit_test.py [hunk 3]
```diff
@@ -141,6 +142,7 @@ def test_url_protocol_fix():
         expected = "'https://flickr.com'"
 
         bad_http = "neverssl.com"
+        TlsTest.test_tls_supported = MagicMock(return_value=False)
```
Comment (krysal) [inline ingestion_server/test/unit_test.py:145]: Nice! It is a more robust test now.

**Review by krysal — COMMENTED** [review]
Django has a command for [squashing migrations](https://docs.djangoproject.com/en/3.2/topics/migrations/#squashing-migrations), didn't it work here? Altering the old ones manually is not a good practice. Otherwise, it seems to work well.

**Comment by dhruvkb** [comment]
@krysal the reason I did not squash the migrations and rather edit them by hand was that this approach removes the need to actually run any migrations in production. 

Editing migrations is bad but, in our case, the prod environment and dev are quite out of sync so the usual guidelines need to be bent a little to bring all environments to the same level again.

**Review by krysal — APPROVED** [review]

**Comment by AetherUnbound** [comment]
This migration ran successfully on a snapshot of prod :heavy_check_mark:
